dpe-stt-proxy/infrastructure/src/queue.py

```diff
@@ -0,0 +1,109 @@
+import sys
+from troposphere import GetAtt, Ref, Template, Join, Parameter
+from troposphere.sqs import Queue, QueuePolicy
+
+t = Template()
+
+t.set_description("AWS CloudFormation Sample Template: This template "
+                  "demonstrates the creation of a DynamoDB table.")
+
+component = t.add_parameter(Parameter(
+    "ComponentName",
+    Type="String",
+    Description=("Component name")
+))
+
+sttGatewayIspyTopic = t.add_parameter(Parameter(
+    "SttGatewayIspyTopic",
+    Type="String",
+    Description=("SttGatewayIspyTopic")
+))
+
+sttBridgeAwsIspyTopic = t.add_parameter(Parameter(
+    "SttBridgeAwsIspyTopic",
+    Type="String",
+    Description=("SttBridgeAwsIspyTopic")
+))
+
+sttIspyConformerIspyTopic = t.add_parameter(Parameter(
+    "SttIspyConformerIspyTopic",
+    Type="String",
+    Description=("SttIspyConformerIspyTopic")
+))
+
+sttNormaliserAwsIspyTopic = t.add_parameter(Parameter(
+    "SttNormaliserAwsIspyTopic",
+    Type="String",
+    Description=("SttNormaliserAwsIspyTopic")
+))
+
+sttOutputNotification = t.add_parameter(Parameter(
+    "SttOutputNotification",
+    Type="String",
+    Description=("SttOutputNotification")
+))
+
+environment = t.add_parameter(Parameter(
+    "Environment",
+    Default="int",
+    Type="String",
+    Description=("Environment name to check against int, test, or live")
+))
+
+recvWaitSeconds = t.add_parameter(Parameter(
+    "ReceiveMessageWaitTimeSeconds",
+    Default="20",
+    Type="Number"
+))
+
+visTimeout = t.add_parameter(Parameter(
+    "VisibilityTimeout",
+    Default="30",
+    Type="Number"
+))
+
+sqsQueue = t.add_resource(Queue(
+    "SQSQueue",
+    QueueName=Join("-", [Ref(environment), Ref(component), 'queue']),
+    ReceiveMessageWaitTimeSeconds=Ref(recvWaitSeconds),
+    VisibilityTimeout=Ref(visTimeout)
+))
+
+t.add_resource(QueuePolicy(
+    "AllowSNS2SQSPolicy",
+    Queues=[Ref(sqsQueue)],
+    PolicyDocument={
+        "Version": "2008-10-17",
+        "Id": "PublicationPolicy",
+        "Statement": [{
+            "Sid": "PSTT-statement",
+            "Effect": "Allow",
+            "Principal": {
+              "AWS": "*"
```

Comment:
Are there any bad side effects from this - presume it's just to have to avoid getting `ec2.amazonaws.com` etc. right every time?

Reply:
I think this was already reviewed at some earlier point in time (need to find the issue..). I'm going to resolve this as it has been discussed and evaluated.